clang/docs/ReleaseNotes.rst

```diff
@@ -214,6 +214,8 @@ Improvements to Clang's diagnostics
   :doc:`ThreadSafetyAnalysis` still does not perform alias analysis. The
   feature will be default-enabled with ``-Wthread-safety`` in a future release.
 
+- Improve the diagnostics for shadows template parameter.
```

Comment:
This should probably mention the bug# as well as slightly better clarify that it is giving a more correct location.